workers/data_refinery_workers/processors/test_smasher.py

```diff
@@ -0,0 +1,348 @@
+    SampleResultAssociation,
+    Dataset,
+    ProcessorJobDatasetAssociation
+)
+from data_refinery_workers.processors import smasher
+from data_refinery_workers.processors import utils
+
+def prepare_job():
+    pj = ProcessorJob()
+    pj.pipeline_applied = "SMASHER"
+    pj.save()
+
+    experiment = Experiment()
+    experiment.accession_code = "GSE51081"
+    experiment.save()
+
+    result = ComputationalResult()
+    result.save()
+
+    homo_sapiens = Organism.get_object_for_name("HOMO_SAPIENS")
+
+    sample = Sample()
+    sample.accession_code = 'GSM1237810'
+    sample.title = 'GSM1237810'
+    sample.organism = homo_sapiens
+    sample.save()
+
+    sra = SampleResultAssociation()
+    sra.sample = sample
+    sra.result = result
+    sra.save()
+
+    computed_file = ComputedFile()
+    computed_file.filename = "GSM1237810_T09-1084.PCL"
+    computed_file.absolute_file_path = "/home/user/data_store/PCL/" + computed_file.filename
+    computed_file.result = result
+    computed_file.size_in_bytes = 123
+    computed_file.save()
+
+    sample = Sample()
+    sample.accession_code = 'GSM1237812'
+    sample.title = 'GSM1237812'
+    sample.organism = homo_sapiens
+    sample.save()
+
+    sra = SampleResultAssociation()
+    sra.sample = sample
+    sra.result = result
+    sra.save()
+
+    computed_file = ComputedFile()
+    computed_file.filename = "GSM1237812_S97-PURE.PCL"
+    computed_file.absolute_file_path = "/home/user/data_store/PCL/" + computed_file.filename
+    computed_file.result = result
+    computed_file.size_in_bytes = 123
+    computed_file.save()
+
+    ds = Dataset()
+    ds.data = {'GSE51081': ['GSM1237810', 'GSM1237812']}
+    ds.aggregate_by = 'EXPERIMENT' # [ALL or SPECIES or EXPERIMENT]
+    ds.scale_by = 'STANDARD' # [NONE or MINMAX or STANDARD or ROBUST]
+    ds.email_address = "null@derp.com"
+    #ds.email_address = "miserlou+heyo@gmail.com"
+    ds.save()
+
+    pjda = ProcessorJobDatasetAssociation()
+    pjda.processor_job = pj
+    pjda.dataset = ds
+    pjda.save()
+
+    return pj
+
+class SmasherTestCase(TestCase):
+
+    @tag("smasher")
+    def test_smasher(self):
+        """ Main tester. """
+        job = prepare_job()
+
+        relations = ProcessorJobDatasetAssociation.objects.filter(processor_job=job)
+        dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+        job_context_check = {}
+        job_context_check["dataset"] = dataset
+        job_context_check["samples"] = dataset.get_samples()
+        job_context_check["experiments"] = dataset.get_experiments()
+        self.assertEqual(len(job_context_check['samples']), 2)
+        self.assertEqual(len(job_context_check['experiments']), 1)
+
+        # Smoke test while we're here..
+        dataset.get_samples_by_experiment()
+        dataset.get_samples_by_species()
+        dataset.get_aggregated_samples()
+
+        # XXX: agg_type 'SPECIES' hangs on Linux, not OSX.
+        # Don't know why yet.
```

Comment:
Is this still true? It looks like it's working on CCI now.

Reply:
It's working on CCI because this is commented out. I don't know why this test causes problems on Linux. :[[[[

Reply:
Womp womp